packages/host/tests/acceptance/code-submode/playground-test.gts

```diff
@@ -995,4 +1199,412 @@ export class BlogPost extends CardDef {
+      assert
+        .dom(
+          '[data-test-field-preview-card] [data-test-compound-field-format="edit"]',
+        )
+        .exists();
+      assert
+        .dom(
+          '[data-test-field-preview-card] [data-test-field="firstName"] input',
+        )
+        .hasNoValue();
+      await fillIn('[data-test-field="firstName"] input', 'Marco');
+      await fillIn('[data-test-field="lastName"] input', 'N.');
+
+      await click('[data-test-instance-chooser]');
+      await click('[data-test-choose-another-instance]');
+      assert
+        .dom('[data-test-field-chooser] [data-test-field-instance]')
+        .exists({ count: 1 });
+      assert
+        .dom('[data-test-field-chooser] [data-test-full-name-embedded]')
+        .hasText('Marco N.');
+      await click('[data-test-field-chooser] [data-test-close-modal]');
+
+      await click('[data-test-accordion-item="spec-preview"] button');
+      assert
+        .dom('[data-test-contains-many="containedExamples"] [data-test-item]')
+        .exists({ count: 1 });
+      assert
+        .dom(
+          '[data-test-contains-many="containedExamples"] [data-test-item="0"] [data-test-field="firstName"] input',
+        )
+        .hasValue('Marco');
+
+      let selection = getPersistedPlaygroundSelection(
+        `${testRealmURL}author/FullNameField`,
+      );
+      assert.deepEqual(selection, {
+        cardId: `${testRealmURL}Spec/full-name`,
+        format: 'edit',
+        fieldIndex: 0,
+      });
+    });
+
+    test('editing compound field instance live updates the preview', async function (assert) {
+      const updatedCommentField = `import { contains, field, Component, FieldDef } from "https://cardstack.com/base/card-api";
+        import StringField from "https://cardstack.com/base/string";
+
+        export class Comment extends FieldDef {
+          static displayName = 'Comment';
+          @field title = contains(StringField);
+          @field name = contains(StringField);
+          @field message = contains(StringField);
+
+          static embedded = class Embedded extends Component<typeof this> {
+        <template>
+          <div data-test-embedded-comment>
+            <p><@fields.message /> - by <@fields.name /></p>
+          </div>
+        </template>
+          }
+        }`;
+      await visitOperatorMode({
+        stacks: [],
+        submode: 'code',
+        codePath: `${testRealmURL}blog-post.gts`,
+      });
+      await selectClass('Comment');
+      await click('[data-test-accordion-item="playground"] button');
+      assert
+        .dom('[data-test-embedded-comment-title]')
+        .hasText('Terrible product');
+      await realm.write('blog-post.gts', updatedCommentField),
+        await waitUntil(
+          () =>
+            document.querySelector('[data-test-embedded-comment-title]') ===
+            null,
+        );
+      assert.dom('[data-test-embedded-comment-title]').doesNotExist();
+    });
+  });
```

Comment:
Is this file getting to the point where it should be broken up? I don‘t know that we have a metric for that but it might be time for a subdirectory 😯

Reply:
Ha good point. I can separate the modules into different files in a subdir